go/client/chat_utils.go

```diff
@@ -0,0 +1,84 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+)
+
+type uidUsernameMapper map[keybase1.UID]string
+
+func (m uidUsernameMapper) getUsername(ctx context.Context, g *libkb.GlobalContext, uid keybase1.UID) (string, error) {
+	if m == nil {
+		m = make(uidUsernameMapper)
+	}
+
+	if username, ok := m[uid]; ok {
+		return username, nil
+	}
+
+	userClient, err := GetUserClient(g)
+	if err != nil {
+		return "", err
+	}
+	var ret keybase1.User
+	if ret, err = userClient.LoadUser(ctx, keybase1.LoadUserArg{
+		Uid: uid,
+	}); err != nil {
+		return "", err
+	}
+
+	m[uid] = ret.Username
+	return ret.Username, err
+}
+
+// parseDurationExtended is like time.ParseDuration, but adds "d" unit. "1d" is
+// one day, defined as 24*time.Hour. Only whole days are supported for "d"
+// unit, but it can be followed by smaller units, e.g., "1d1h".
+func parseDurationExtended(s string) (d time.Duration, err error) {
```

Comment:
Right, was thinking along those lines, but there are lots of subtle usability problems that come up with that, talking them through with @malgorithms. For instance: enums are they ints or strings?  Or key names, are they snake-cased or command-line style or camelcase?